core/src/interfaces.rs

```diff
@@ -442,14 +442,84 @@ impl TryFrom<ProofRequestOpt> for ProofRequest {
     }
 }
 
-#[serde_as]
-#[derive(Clone, Debug, Serialize, Deserialize)]
+#[derive(Default, Clone, Serialize, Deserialize, Debug, ToSchema)]
+#[serde(default)]
 /// A request for proof aggregation of multiple proofs.
 pub struct AggregationRequest {
-    /// All the proofs to verify
-    pub proofs: Vec<Proof>,
+    /// The block numbers and l1 inclusion block numbers for the blocks to aggregate proofs for.
+    pub block_numbers: Vec<(u64, u64)>,
```

Comment:
I also added this in the design document, but doesn't it make more sense for this to just take a list of proofs? 

https://www.notion.so/Proof-aggregation-API-dbd4fcec46d049b78132072555386a40?d=6052f3ea6244452e98add34ec292cd1a&pvs=4#05a93b35e0e941c7b94ef8bed5be6112

Reply:
cp the notion reply and let's discuss it here.

> Yes, this is what I called streaming style, raiko generating proofs at the same pace as before, but report aggregated one if it is asked for aggregation proof. better for runtime, less burst resource requirement, but bit complex-er status management. 
We talked that with client before, they thought the list is easier for initial test, less effort in dev, so we decided to fast impl this to let aggregation go devnet quickly. But I think that streaming style will eventually replace it.

I think the streaming style is what you want, but perhaps I mis-understand your optimistic mode, correct me if so.

Reply:
@Brechtpd It does make more sense to keep it as a separate route to only aggregate existing proofs, but as @smtmfft and I talked about, for the initial version it is easier for the client to test it with a similar request style which proves all the requested blocks.
I will create a aggregate-only endpoint to handle the optimal scenario, but for now just allowing a quicker path to integration unless we want to go with the more optimal approach right away.

Reply:
If there are other considerations than of course okay with me with the current approach, just trying to avoid any additional work if going from the current block indices to the proofs one would be significantly different and so some code being written now would be deprecated quite quickly.